src/features/PowerShellFindModule.ts

```diff
@@ -20,28 +20,35 @@ export class FindModuleFeature implements IFeature {
 
     private command: vscode.Disposable;
     private languageClient: LanguageClient;
+    private cancelFindToken: vscode.CancellationTokenSource;
 
     constructor() {
         this.command = vscode.commands.registerCommand('PowerShell.PowerShellFindModule', () => {
-            var items: QuickPickItem[] = [];
+            // It takes a while to get the list of PowerShell modules, display some UI to let user know
+            this.cancelFindToken = new vscode.CancellationTokenSource();
+            vscode.window
+                .showQuickPick(
+                ["Cancel"],
+                { placeHolder: "Please wait, retrieving list of PowerShell modules. This can take some time..." },
+                this.cancelFindToken.token)
+                .then(response => { if (response === "Cancel") { this.clearCancelFindToken(); } });
+
+            // Cancel the loading prompt after 60 seconds
+            setTimeout(() => {
+                if (this.cancelFindToken) {
+                    this.clearCancelFindToken();
+
+                    vscode.window.showErrorMessage(
+                        "The online source for PowerShell modules is not responding. Cancelling Find/Install PowerShell command.");
+                }
+            }, 60000);
 
-            vscode.window.setStatusBarMessage(this.getCurrentTime() + " Initializing...");
-            this.languageClient.sendRequest(FindModuleRequest.type, null).then((modules) => {
-                for(var item in modules) {
-                    items.push({ label: modules[item].name, description: modules[item].description });
-                };
-
-                vscode.window.setStatusBarMessage("");
-                Window.showQuickPick(items,{placeHolder: "Results: (" + modules.length + ")"}).then((selection) => {
-                    if (!selection) { return; }
-                    switch (selection.label) {
-                        default :
-                            var moduleName = selection.label;
-                            //vscode.window.setStatusBarMessage("Installing PowerShell Module " + moduleName, 1500);
-                            this.languageClient.sendRequest(InstallModuleRequest.type, moduleName);
-                        }
-                    });
-                });
+            this.pickPowerShellModule().then((moduleName) => {
+                if (moduleName) {
+                    // vscode.window.setStatusBarMessage("Installing PowerShell Module " + moduleName, 1500);
+                    this.languageClient.sendRequest(InstallModuleRequest.type, moduleName);
```

Comment:
I left this comment in.  I think the thought was it would be nice to display this status bar message but the extension, should probably receive a message back from PSES that the module has been installed.  Then it would display an "Install complete" message for say 5 seconds before being cleared.